enhancements/service-catalog/prepare-service-catalog-and-brokers-for-removal.md

```diff
@@ -0,0 +1,173 @@
+--
+title: prepare-service-catalog-and-brokers-for-removal
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+approvers:
+  - TBD
+creation-date: 2019-09-12
+last-updated: 2019-09-16
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# prepare-service-catalog-and-brokers-for-removal
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Operators became the strategy in OpenShift 4.x for customers to manage their
+applications in the cluster. As this technology has improved, the need for
+having a Service Catalog backed by Service Brokers is no longer needed. As an
+enterprise product, we need to deprecate the catalog and brokers, inform the
+users of this deprecation, and then finally remove them from the product in
+OpenShift 4.4. This should allow plenty of time to migrate any broker related
+items the customer may have had to Operators.
+
+## Motivation
+
+As our strategy shifted from Service Brokers to Operators, customers and
+partners need sufficient time to shift their strategy towards Operators as well.
+
+### Goals
+
+Service Catalog, Service Brokers and their associated operators need to be
+deprecated and removed by OpenShift 4.4.
+
+Alert users that the above have been deprecated and will be removed in a
+future OpenShift release.
+
+### Non-Goals
+
+Not in scope is writing the preflight-check for CVO.
+
+## Proposal
+
+### User Stories [optional]
+
+N/A or TBD
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+The 3 operators, Service Catalog Controller Manager operator (svcat-cm-op),
+the Ansible Service Broker Operator (ASBO) and the Template Service Broker
+Operator (TSBO), will need a metric to tell folks when any of these are in use.
+
+Once a ServiceMonitor is created for Prometheus, so the user and telemeter can
+see the alert of when someone has enabled the catalog or brokers.
```

Comment:
Can you expand on this a little bit?
* What metric do we plan to add for each operator?
  * cluster-svcat-apiserver-operator already [defines](https://github.com/openshift/cluster-svcat-apiserver-operator/blob/4eade52a42c9a71fc714c6e32212c762cd9bd1cd/pkg/version/version.go#L37-L43) `openshift_cluster_openshift_apiserver_operator_build_info`
  * cluster-svcat-controller-manager-operator already [defines](https://github.com/openshift/cluster-svcat-controller-manager-operator/blob/77b65f8164288694bddd4a4ded7a81dc2a1c70f0/pkg/version/version.go#L37-L43) `openshift_service_serving_cert_signer_build_info` (**That seems like it needs to be fixed**)
  * ASBO and TSBO are ansible-based operators. Do they have the necessary extension points to add these metrics, or can the built-in metrics be re-used?
* Is each Operator responsible for creating its own ServiceMonitor?
  * cluster-svcat-apiserver-operator already [defines](https://github.com/openshift/cluster-svcat-apiserver-operator/blob/292e1eacfca31ae94fb2375a2a435c4994f79c06/manifests/0000_90_cluster-svcat-apiserver-operator_02-operator-servicemonitor.yaml) one
  * cluster-svcat-controller-manager-operator already [defines](https://github.com/openshift/cluster-svcat-controller-manager-operator/blob/f8ad3be67ae5baaa8c2c5d167ba48611146df560/bindata/v3.11.0/openshift-svcat-controller-manager/servicemonitor.yaml) one 
 * Is each Operator responsible for creating its own Alert?

Do we have enough information at this point to break this down into user stories and tasks?